ui/src/shared/actions/sources.js

```diff
@@ -1,35 +1,30 @@
-export function loadSources(sources) {
-  return {
-    type: 'LOAD_SOURCES',
-    payload: {
-      sources,
-    },
-  };
-}
+import {deleteSource, getSources} from 'src/shared/apis'
 
-export function updateSource(source) {
-  return {
-    type: 'SOURCE_UPDATED',
-    payload: {
-      source,
-    },
-  };
-}
+export const loadSources = (sources) => ({
+  type: 'LOAD_SOURCES',
+  payload: {
+    sources,
+  },
+})
 
-export function removeSource(source) {
-  return {
-    type: 'SOURCE_REMOVED',
-    payload: {
-      source,
-    },
-  };
-}
+export const updateSource = (source) => ({
+  type: 'SOURCE_UPDATED',
+  payload: {
+    source,
+  },
+})
+
+export const addSource = (source) => ({
+  type: 'SOURCE_ADDED',
+  payload: {
+    source,
+  },
+})
+
+// Async action creators
 
-export function addSource(source) {
-  return {
-    type: 'SOURCE_ADDED',
-    payload: {
-      source,
-    },
-  };
+export const removeAndLoadSources = (source) => async (dispatch) => {
+  await deleteSource(source)
+  const sources = await getSources()
```

Comment:
syntactic convention here has been to peel off whatever data, ex. 'const {data}' or even 'const {data: {sources}}'